ui/tx/Noves/components/NovesTokensCard.tsx

```diff
@@ -0,0 +1,54 @@
+import { Box, Text, useColorModeValue } from '@chakra-ui/react';
+import type { FC } from 'react';
+import React from 'react';
+
+import type { NovesNft, NovesToken } from 'types/novesApi';
+
+import CopyToClipboard from 'ui/shared/CopyToClipboard';
+
+interface Props {
+  amount?: string;
+  token: NovesToken | NovesNft | undefined;
+}
+
+const NovesTokensCard: FC<Props> = ({ token, amount }) => {
+  const textColor = useColorModeValue('white', 'blackAlpha.900');
+
+  if (!token) {
+    return null;
+  }
+
+  const showTokenName = token.symbol !== token.name;
+  const showTokenAddress = Boolean(token.address.match(/^0x[a-fA-F\d]{40}$/));
```

Comment:
Are there cases where the token address is not in the form of a hash?

Reply:
Not really on our side. When the asset is the native coin the address won't be a hash, but there's no real "address" concept at that point. So it makes sense not to display the address for "ETH" because that doesn't really exist.